concourse: add pool codecommit repo per team

tlwr opened the pull request:
What
----

To enable teams to use the pool-resource, they need:

- a git repo
- an ssh key
- etc

this is a bit annoying to set up, as such, no one is using pools

we can terraform these for each team, so they don't have to do any work to use pools

How to review
----

We made this pipeline:

```
resource_types:
  - name: pool
    type: registry-image
    source:
      repository: concourse/pool-resource
      tag: 1.1.3

resources:
  - name: my-pool-repo
    type: git
    source:
      branch: pool
      uri: ((readonly_codecommit_pool_uri))
      private_key: ((readonly_codecommit_private_key))

  - name: a-lock-config
    type: mock
    source:
      create_files:
        name: my-lock
        metadata: ''

  - name: my-pool
    type: pool
    source:
      branch: pool
      uri: ((readonly_codecommit_pool_uri))
      private_key: ((readonly_codecommit_private_key))
      pool: my-pool

jobs:
  - name: recreate-pool
    plan:
      - task: create-pool-dirs
        config:
          platform: linux

          image_resource:
            type: registry-image
            source:
              repository: governmentpaas/git-ssh
              tag: latest

          outputs:
            - name: my-pool-repo

          run:
            dir: my-pool-repo
            path: sh
            args:
              - -c
              - |
                set -eu

                git init .

                mkdir -p my-pool/claimed
                mkdir -p my-pool/unclaimed
                touch my-pool/claimed/.keep
                touch my-pool/unclaimed/.keep

                git add my-pool
                git commit -m 'setup my-pool'

      - put: my-pool-repo
        params:
          repository: my-pool-repo
          force: true

  - name: create-lock
    plan:
      - get: a-lock-config

      - put: my-pool
        params:
          add: a-lock-config

  - name: acquire-lock
    plan:
      - get: my-pool
        passed: [create-lock]

      - put: my-pool
        params:
          acquire: true

  - name: release-lock
    plan:
      - get: my-pool
        passed: [create-lock]

      - put: my-pool
        params:
          release: my-pool

```

and deployed it to staging: https://cd-staging.gds-reliability.engineering/teams/sandbox/pipelines/pool-test

Commit:
concourse: add pool codecommit repo per team

To enable teams to use the pool-resource, they need:

- a git repo
- an ssh key
- etc

this is a bit annoying to set up, as such, no one is using pools

we can terraform these for each team, so they don't have to do any work
to use pools

Signed-off-by: Toby Lorne <toby.lornewelch-richards@digital.cabinet-office.gov.uk>
Co-authored-by: Aditya <aditya.pahuja@digital.cabinet-office.gov.uk>

tlwr commented:
Also https://github.com/alphagov/tech-ops-private/pull/380

chrisfarms commented:
Looks useful :+1: 

Couple of thoughts:

* It looks like this adds more things to remember when setting up a new team ... Could we add a line to [the bit in the manual that keeps track of these things](https://re-team-manual.cloudapps.digital/automate-on-call-interruptible.html#creating-a-new-team)
* Is it possible to use the pool-resource in the deploy pipeline as a simple test against regressions?

Commit:
concourse: test pool resource in deployer pipeline

Signed-off-by: Toby Lorne <toby.lornewelch-richards@digital.cabinet-office.gov.uk>

Commit:
concourse: tag release is own job

when testing pipeline, nice to have separate tag release job, so it can
be paused without pausing test job

Signed-off-by: Toby Lorne <toby.lornewelch-richards@digital.cabinet-office.gov.uk>

Commit:
concourse: deploy github icons exist

Signed-off-by: Toby Lorne <toby.lornewelch-richards@digital.cabinet-office.gov.uk>

tlwr commented:
Refactorino'd the pipeline a bit
this is deployed now: https://cd-staging.gds-reliability.engineering/teams/main/pipelines/deploy

Once approved I'll change the tracked pipeline back to master and then merge and unpause

chrisfarms commented on reliability-engineering/pipelines/concourse-deployer.yml:
```diff
@@ -64,6 +71,27 @@ resources:
+    private_key: ((readonly_codecommit_private_key))
+- name: pool
+  type: pool
+  icon: pool
```
:swimmer:

tlwr replied:
concourse-pool-party.jpeg
